.github/workflows/docker.yml

```diff
@@ -0,0 +1,200 @@
+            echo build-cache="type=local,src=/tmp,dest=/tmp,mode=max" >>"$GITHUB_OUTPUT"
+          else
+            build_cache="type=registry,image-manifest=true,oci-mediatypes=true,ref=$image_repository:cache"
+            echo build-cache="$build_cache" >>"$GITHUB_OUTPUT"
+            echo build-cache-shared="$build_cache" >>"$GITHUB_OUTPUT"
+          fi
+
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        if: ${{ !github.event.localrun && env.IMAGE_REPOSITORY }}
+        id: aws-creds
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-session-name: container-build
+          aws-region: ${{ vars.AWS_REGION }}
+          # Identical to usage in push.yml
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          # TODO: Use this instead of aws-{access,secret}, but needs proper roles & policies
+          # role-to-assume: ${{ vars.AWS_ECR_ROLE }}
+
+      - name: Log in to Amazon ECR
+        if: ${{ steps.aws-creds.conclusion == 'success' }}
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker-container
+          # driver-opts: image=moby/buildkit:buildx-stable-1
+          driver-opts: |
+            image=${{vars.AWS_ECR_REPO_BASE}}/moby/buildkit:buildx-stable-1
+          install: true
+          use: true
+
+      - name: Generate image metadata
+        id: meta
+        # This step requires a valid GitHub token to query the API for e.g. description of repository
+        if: ${{ !github.event.localrun }}
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ steps.inputs.outputs.image-repository }}
+          tags: |
+            # Edge without prefix should only be used on `main`, so as not to conflict with type=sha
+            # type=edge
+            type=edge,prefix=edge-
+            # SemVer by tag (e.g. v1.2.3)
+            type=semver,pattern={{version}}
+            type=pep440,pattern={{version}}
+            # PR branch/name
+            type=ref,event=branch
+            type=ref,event=push
+            type=ref,event=pull_request
+            type=ref,event=workflow_dispatch
+            # Git SHA
+            type=sha,format=short,prefix=
+            type=sha,format=long,prefix=
+            # The final sha for the merge commit
+            # mcgh: Merge Commit GitHub
+            type=raw,value=mcgh-${{ github.event.pull_request.merge_commit_sha }}
+
+      - name: Docker build/cache dependencies
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: scripts/ci/Dockerfile
+          push: false
+          target: deps
+          build-args: |
+            NODE_IMAGE_TAG=${{ needs.prepare.outputs.node-image-tag }}
```

Comment:
see changes in the [Dockerfile](https://github.com/island-is/island.is/blob/main/scripts/ci/Dockerfile#L3)

```
NODE_IMAGE_VERSION
```